### PR TITLE
Feat/init codeceptjs with ts

### PR DESCRIPTION
**Pull request by kobenguyent**
## Motivation/Description of the PR
- This change gives you the possibility to init codeceptjs with typescript
```
thanh@Thanh-MBP14 automation % npx codeceptjs init

  Welcome to CodeceptJS initialization tool
  It will prepare and configure a test environment for you

Installing to /Users/thanh/Desktop/automation
? Is your project Typescript? yes
? Where are your tests located? ./*_test.js
? What helpers do you want to use? REST
? Where should logs, screenshots, and reports to be stored? ./output
? Do you want localization for tests? (See https://codecept.io/translation/) English (no localization)
Configure helpers...
? [REST] Endpoint of API you are going to test http://localhost:3000/api

Steps file created at ./steps_file.ts
Config created at /Users/thanh/Desktop/automation/codecept.conf.ts
Directory for temporary output files is already created at './output'
tsconfig.json already exists at /Users/thanh/Desktop/automation/tsconfig.json
jsconfig.json already exists at /Users/thanh/Desktop/automation/jsconfig.json
TypeScript Definitions provide autocompletion in Visual Studio Code and other IDEs
Definitions were generated in steps.d.ts

 Almost ready... Next step:
Creating a new test...
----------------------
? Feature which is being tested (ex: account, login, etc) 
? Filename of a test _test.js

Test for _test.js was created in /Users/thanh/Desktop/automation/_test.js

--
CodeceptJS Installed! Enjoy supercharged testing! 🤩
Find more information at https://codecept.io


--
Please install dependent packages locally: npm install --save-dev typescript ts-node

```

## Type of change
- [x] :rocket: New functionality

## Checklist:
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Review by Arhell — APPROVED** [review]

**Review by DavertMik — CHANGES_REQUESTED** [review]
I really like the idea but it should be improved:

> Filename of a test _test.js

Tests should be created with `ts` extension
As well as config should be `codecept.conf.ts`

**Comment by kobenguyent** [comment]
> I really like the idea but it should be improved:
> 
> > Filename of a test _test.js
> 
> Tests should be created with `ts` extension As well as config should be `codecept.conf.ts`

Updated as requested.

**Comment by kobenguyent** [comment]
Not sure why the Run Unit tests / build (12.x) failed, I tried on my local, there is no issue
```
> codeceptjs@3.3.4 test:runner
> mocha test/runner --recursive



  CodeceptJS Timeouts
    ✓ should stop test when timeout exceeded (5434ms)
    ✓ should take --no-timeouts option (6307ms)
    ✓ should ignore timeouts if no timeout (1301ms)
    ✓ should use global timeouts if timeout is set (1283ms)
    ✓ should prefer step timeout (3376ms)
    ✓ should keep timeout with steps (392ms)


  6 passing (18s)


```

**Review by EgorBodnarUpteam — APPROVED** [review]

File: lib/command/init.js [hunk 1]
```diff
@@ -80,7 +87,19 @@ module.exports = function (initPath) {
...
+    error(`Config is already created at ${typeScriptconfigFile}`);
+    return;
+  }
+
   inquirer.prompt([
+    {
+      name: 'typescript',
+      type: 'list',
+      choices: ['No', 'Yes'],
+      message: 'Is your project Typescript?',
```
Comment (DavertMik) [inline lib/command/init.js]:
```suggestion
      message: 'Do you plan to use TypeScript?',
```

File: lib/command/init.js [hunk 2]
```diff
@@ -105,11 +124,23 @@ module.exports = function (initPath) {
...
+      configHeader = tsNodeRequired + configHeader;
+    }
+
     const config = defaultConfig;
     config.name = testsPath.split(path.sep).pop();
     config.output = result.output;
 
     config.tests = result.tests;
+    if (isTypeScript) {
+      config.tests = `${config.tests.replace(/js/, extension)}`;
```
Comment (DavertMik) [inline lib/command/init.js]:
```suggestion
      config.tests = `${config.tests.replace(/\.js$/, '.' + extension)}`;
```

File: lib/command/init.js [hunk 3]
```diff
@@ -80,7 +87,19 @@ module.exports = function (initPath) {
...
+  const typeScriptconfigFile = path.join(testsPath, 'codecept.conf.ts');
+  if (fileExists(typeScriptconfigFile)) {
+    error(`Config is already created at ${typeScriptconfigFile}`);
+    return;
+  }
+
   inquirer.prompt([
+    {
+      name: 'typescript',
+      type: 'list',
```
Comment (DavertMik) [inline lib/command/init.js]: image

Please try to use `confirm` option here

**Review by DavertMik — CHANGES_REQUESTED** [review]

File: lib/command/init.js [hunk 4]
```diff
@@ -80,7 +87,19 @@ module.exports = function (initPath) {
...
+  if (fileExists(typeScriptconfigFile)) {
+    error(`Config is already created at ${typeScriptconfigFile}`);
+    return;
+  }
+
   inquirer.prompt([
+    {
+      name: 'typescript',
+      type: 'confirm',
+      choices: ['No', 'Yes'],
```
Comment (DavertMik) [inline lib/command/init.js]: did you try this actually?
does confirm option actually needs choices?

Comment (kobenguyent) [reply lib/command/init.js]: Ah no. Will look closely then.

**Comment by DavertMik** [comment]
Thanks, @PeterNgTr 
If you tried it and it works for you I'm merging

**Comment by kobenguyent** [comment]
> Thanks, @PeterNgTr 
> 
> If you tried it and it works for you I'm merging

Yeah, just gave it a try. Working like a charm.

**Pull request merged**